staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder_test.go

```diff
@@ -417,6 +421,95 @@ func TestNewBuilder(t *testing.T) {
 	}
 }
 
+func TestRouteParameterBuilder(t *testing.T) {
+	testCRDKind := "Foo"
+	testCRDGroup := "foo-group"
+	testCRDVersion := "foo-version"
+
+	testCases := []struct {
+		name        string
+		crd         *apiextensions.CustomResourceDefinition
+		version     string
+		expectedErr error
+	}{
+		{
+			name:    "building normal namespaced crd should work",
+			version: testCRDVersion,
+			crd: &apiextensions.CustomResourceDefinition{
+				Spec: apiextensions.CustomResourceDefinitionSpec{
+					Group: testCRDGroup,
+					Names: apiextensions.CustomResourceDefinitionNames{
+						Kind: testCRDKind,
+					},
+					Scope: apiextensions.NamespaceScoped,
+					Versions: []apiextensions.CustomResourceDefinitionVersion{
+						{
+							Name:    testCRDVersion,
+							Storage: true,
+							Served:  true,
+							Schema:  &apiextensions.CustomResourceValidation{},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "building normal cluster-scope crd should work",
+			version: testCRDVersion,
+			crd: &apiextensions.CustomResourceDefinition{
+				Spec: apiextensions.CustomResourceDefinitionSpec{
+					Group: testCRDGroup,
+					Names: apiextensions.CustomResourceDefinitionNames{
+						Kind: testCRDKind,
+					},
+					Scope: apiextensions.ClusterScoped,
+					Versions: []apiextensions.CustomResourceDefinitionVersion{
+						{
+							Name:    testCRDVersion,
+							Storage: true,
+							Served:  true,
+							Schema:  &apiextensions.CustomResourceValidation{},
+						},
+					},
+				},
+			},
+		},
+	}
+	verifyPath := func(pathName string, path spec.PathItem, shouldHaveNamespaceParam, shouldHaveNameParam bool) error {
+		hasNameParam := false
+		hasNamespaceParam := false
+		for _, param := range path.Parameters {
+			if param.In == "path" && param.Name == "namespace" {
+				hasNamespaceParam = true
+			}
+			if param.In == "path" && param.Name == "name" {
+				hasNameParam = true
+			}
+		}
+		if shouldHaveNamespaceParam && !hasNamespaceParam {
```

Comment:
this doesn't catch a cluster-scoped CRD incorrectly getting a namespace parameter added

Reply:
updated